### PR TITLE
fix: Fixing sonarcloud metrics on dashboard

### PR DESCRIPTION
**Pull request by rakatyal**
This issue started on [this run](https://github.com/sendgrid/dx-automator/actions/runs/3394380380/jobs/5642868964) (compared with the [previous run](https://github.com/sendgrid/dx-automator/actions/runs/3346049480/jobs/5542362765)) where the version of python flipped to 3.11

The tests also started failing on python version [3.11](https://github.com/sendgrid/dx-automator/actions/runs/3579623000/jobs/6021062729#step:5:10) vs compared to the previous [3.10](https://github.com/sendgrid/dx-automator/actions/runs/3372055459/jobs/5595035917#step:5:10) one.

This PR adds a .python-version file to pin the python version used for this repo.

### CODE REVIEW

File: terraform/datadog_dashboards.tf [hunk 1]
```diff
@@ -51,15 +51,15 @@ resource "datadog_dashboard" "ordered_dashboard" {
             query {
               metric_query {
                   data_source = "metrics"
-                  query       = "avg:sonar_cloud.measures.lines_to_cover{org:twilio,pre-release:false} by {repo}"
+                  query       = "avg:sonar_cloud.measures.Metrics.LINES_TO_COVER{org:twilio,pre-release:false} by {repo}"
```
Comment (childish-sambino) [inline terraform/datadog_dashboards.tf:54]: This issue started on [this run](https://github.com/sendgrid/dx-automator/actions/runs/3394380380/jobs/5642868964) (compared with the [previous run](https://github.com/sendgrid/dx-automator/actions/runs/3346049480/jobs/5542362765)) where the version of python flipped to 3.11

We should lock down python to a specific version using a `.python-version` file as recommended [here](https://github.com/sendgrid/dx-automator/actions/runs/3346049480/jobs/5542362765) and should see why these tests for the metric collection aren't failing.

Comment (rakatyal) [reply terraform/datadog_dashboards.tf:54]: Yeah I can pin the 3.10 version. Wondering if using the [python-version](https://github.com/actions/setup-python/blob/main/action.yml#L6) input is a better idea or creating the .python-version file? 

The tests also started failing on [3.11](https://github.com/sendgrid/dx-automator/actions/runs/3579623000/jobs/6021062729#step:5:10) vs compared to the previous [3.10](https://github.com/sendgrid/dx-automator/actions/runs/3372055459/jobs/5595035917#step:5:10) one.

Comment (childish-sambino) [reply terraform/datadog_dashboards.tf:54]: I like the file version as it's 1 place to set it which can be used locally as well as by all of the GH workflows.

File: .gitignore [hunk 2]
```diff
@@ -68,9 +68,6 @@ target/
 # IPython Notebook
 .ipynb_checkpoints
 
-# pyenv
-.python-version
```
Comment (childish-sambino) [inline .gitignore:72]: Don't think this should be ignored. That would make it more difficult to update the version later.

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**